In [1]:
# Установить необходимый пакет для работы с URL
install.packages("readr")

# Загрузить библиотеку
library(readr)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [2]:
# Указать URL-адрес файла CSV на GitHub
url <- "https://raw.githubusercontent.com/kindykbayev/KIS/main/lab1/Computers.csv"

In [3]:
Sys.setenv("VROOM_CONNECTION_SIZE" = 1048576)

In [4]:
# Прочитать CSV-файл
data <- read_csv(url, col_types = cols(.default = "integer", cd = col_character(), multi = col_character(), premium = col_character()))

New names:
• `` -> `...1`


In [5]:
View(data)

...1 price speed hd   ram screen cd  multi premium ads trend
1     1   1499  25     80   4  14     no  no    yes     94  1    
2     2   1795  33     85   2  14     no  no    yes     94  1    
3     3   1595  25    170   4  15     no  no    yes     94  1    
4     4   1849  25    170   8  14     no  no    no      94  1    
5     5   3295  33    340  16  14     no  no    yes     94  1    
6     6   3695  66    340  16  14     no  no    yes     94  1    
7     7   1720  25    170   4  14     yes no    yes     94  1    
8     8   1995  50     85   2  14     no  no    yes     94  1    
9     9   2225  50    210   8  14     no  no    yes     94  1    
10   10   2575  50    210   4  15     no  no    yes     94  1    
11   11   2195  33    170   8  15     no  no    yes     94  1    
12   12   2605  66    210   8  14     no  no    yes     94  1    
13   13   2045  50    130   4  14     no  no    yes     94  1    
14   14   2295  25    245   8  14     no  no    yes     94  1    
15   15   2699  50    212   8  14     no  no    yes     94  1    
16   16   2225  50    130   4  14     no  no    yes     94  1    
17   17   1595  33     85   2  14     no  no    yes     94  1    
18   18   2325  33    210   4  15     no  no    yes     94  1    
19   19   2095  33    250   4  15     no  no    yes     94  1    
20   20   4395  66    452   8  14     no  no    yes     94  1    
21   21   1695  33    130   4  14     no  no    yes     94  1    
22   22   2795  66    130   4  14     no  no    yes     94  1    
23   23   2895  25    340  16  14     no  no    yes     94  1    
24   24   2875  50    210   4  17     no  no    yes     94  1    
25   25   4195  50    452   8  14     no  no    yes     94  1    
26   26   1290  33     80   2  14     no  no    yes     94  1    
27   27   1975  33    130   4  14     no  no    yes     94  1    
28   28   3995  33    452   8  14     no  no    yes     94  1    
29   29   3095  33    340  16  14     no  no    yes     94  1    
30   30   3244  66    245   8  14     no  no    yes     94  1    
⋮    ⋮    ⋮     ⋮     ⋮    ⋮   ⋮      ⋮   ⋮     ⋮       ⋮   ⋮    
6230 6230 2344   66    850 16  15     yes no    yes     39  35   
6231 6231 2890  100   1200 24  17     yes no    yes     39  35   
6232 6232 2390   66    850 16  17     yes no    yes     39  35   
6233 6233 1495   66    528  8  14     no  no    yes     39  35   
6234 6234 1844  100    528  8  14     no  no    yes     39  35   
6235 6235 1699   66    630  8  15     no  no    no      39  35   
6236 6236 3290   75   1600 32  17     yes no    yes     39  35   
6237 6237 2790   66   1200 24  17     yes no    yes     39  35   
6238 6238 2845  100   1600 24  15     yes no    yes     39  35   
6239 6239 1744   66    528  8  14     no  no    yes     39  35   
6240 6240 3095  100   1600 32  15     yes no    yes     39  35   
6241 6241 1595  100    528  8  14     no  no    yes     39  35   
6242 6242 2254  100    850 16  15     yes no    yes     39  35   
6243 6243 2723  100   1200 24  15     yes yes   yes     39  35   
6244 6244 2745   66   1600 24  15     yes no    yes     39  35   
6245 6245 3054   75   1600 32  15     yes no    yes     39  35   
6246 6246 2744   66   1200 24  15     yes no    yes     39  35   
6247 6247 1554   66    528  8  14     no  no    yes     39  35   
6248 6248 1654  100    528  8  14     no  no    yes     39  35   
6249 6249 2623   66   1200 24  15     yes yes   yes     39  35   
6250 6250 2095   66    850 16  15     yes no    yes     39  35   
6251 6251 2495   66   1200 24  15     yes no    yes     39  35   
6252 6252 2995   66   1600 32  15     yes no    yes     39  35   
6253 6253 1690   66   1200  8  14     no  no    yes     39  35   
6254 6254 2154   66    850 16  15     yes no    yes     39  35   
6255 6255 1690  100    528  8  15     no  no    yes     39  35   
6256 6256 2223   66    850 16  15     yes yes   yes     39  35   
6257 6257 2654  100   1200 24  15     yes no    yes     39  35   
6258 6258 2195  100    850 16  15     yes no 

In [6]:
set.seed(13)
Count_low<-data.frame()
for (i in c(1:50)){
    Count_low<-rbind(Count_low, c(i,sample(c(3:6),1), sample(c(6:20),1), sample(c(1:20),1), sample(c(1:9),1)))}
colnames(Count_low)<-c("Вариант","Количество С1 >=","Количество С2 >=", "Количество С3 >=", "Количество С4 >=")
View(Count_low[6, ])

Вариант Количество С1 >= Количество С2 >= Количество С3 >= Количество С4 >=
6 6       6                6                11               4

In [7]:
Count_up<-data.frame()
for (i in c(1:50)){
    Count_up<-rbind(Count_up, c(i,Count_low[i,2]+sample(c(3:6),1), Count_low[i,3]+sample(c(6:20),1),
                                          Count_low[i,4]+sample(c(1:20),1), Count_low[i,5]+sample(c(1:9),1)))}
colnames(Count_up)<-c("Вариант","Количество С1 <=","Количество С2 <=", "Количество С3 <=", "Количество С4 <=")
View(Count_up[6, ])

Вариант Количество С1 <= Количество С2 <= Количество С3 <= Количество С4 <=
6 6       10               22               15               13

In [8]:
Res<-lm(price~speed+ram+hd, data)
Res


Call:
lm(formula = price ~ speed + ram + hd, data = data)

Coefficients:
(Intercept)        speed          ram           hd  
  1492.1775       5.8112      78.2357      -0.5357  


In [9]:
g<-predict(Res,data)
View(data.frame(data$price, g))

data.price g       
1    1499       1907.542
2    1795       1794.882
3    1595       1859.328
4    1849       2172.271
5    3295       2753.573
6    3695       2945.343
7    1720       1859.328
8    1995       1893.672
9    2225       2296.121
10   2575       1983.179
11   2195       2218.760
12   2605       2389.100
13   2045       2026.036
14   2295       2132.092
15   2699       2295.050
16   2225       2026.036
17   1595       1794.882
18   2325       1884.388
19   2095       1862.960
20   4395       2259.456
21   1695       1927.246
22   2795       2119.015
23   2895       2707.084
24   2875       1983.179
25   4195       2166.477
26   1290       1797.561
27   1975       1927.246
28   3995       2067.687
29   3095       2753.573
30   3244       2370.350
⋮    ⋮          ⋮       
6230 2344       2672.125
6231 2890       3308.089
6232 2390       2672.125
6233 1495       2218.741
6234 1844       2416.322
6235 1699       2164.098
6236 3290       3574.407
6237 2790       3110.509
6238 2845       3093.801
6239 1744       2218.741
6240 3095       3719.687
6241 1595       2416.322
6242 2254       2869.706
6243 2723       3308.089
6244 2745       2896.221
6245 3054       3574.407
6246 2744       3110.509
6247 1554       2218.741
6248 1654       2416.322
6249 2623       3110.509
6250 2095       2672.125
6251 2495       3110.509
6252 2995       3522.106
6253 1690       1858.737
6254 2154       2672.125
6255 1690       2416.322
6256 2223       2672.125
6257 2654       3308.089
6258 2195       2869.706
6259 2490       2869.706

Существует 4 сегмента в которые требуется выбрать кол-во компьютеров установленное вариантом:
$$6<=С1<=10$$
$$6<=С2<=22$$
$$11<=С3<=15$$
$$4<=С4<=13$$

Стоимость отдельного компьютера от конструктивных параметров  𝑠𝑝𝑒𝑒𝑑,𝑟𝑎𝑚,ℎ𝑑  выражена ранее через регрессионную зависимость:
$$𝑝𝑟𝑖𝑐𝑒=1492.1775+5.8112⋅𝑠𝑝𝑒𝑒𝑑+78.2357⋅𝑟𝑎𝑚−0.5357⋅ℎ𝑑$$

Добавим ограничения нижнего уровня:
 - суммарная стоимость компьютеров сегмента $C_1$ должна быть не более 10000 у.е.
 - суммарная производительность $C_2$ не менее 300 (считаем аддитивной характеристикой от количества входящих ПК)
 - максимальный объем ОЗУ для сегмента $C_3$ не более 8
 - минимальный объем жесткого диска для сегмента для сегмента $C_4$ не менее 1000
 - Суммарная производительность для С4 - не менее 800 (считаем аддитивной характеристикой от количества входящих ПК)

Введём переменные:

$N_1$ - количество серверов

$N_2$ - количество компьютеров сегмента аналитиков и менеджеров

$N_3$ - количество компьютеров сегмента обслуживающего персонала

$N_4$ - количество компьютеров сегмента с видео, аудио оборудованием

$price$ - стоимость компьютера без учёта характеристик

$speed$ - зависимость стоимости компьютера от его скорости

$ram$ - зависимость стоимости компьютера от его оперативной памяти

$hd$ - зависимость стоимости компьютера от объёма диска

Критерий верхнего уровня:

$$𝐻1=𝑁1⋅𝑝𝑟𝑖𝑐𝑒1+𝑁2⋅𝑝𝑟𝑖𝑐𝑒2+𝑁3⋅𝑝𝑟𝑖𝑐𝑒3+𝑁4⋅𝑝𝑟𝑖𝑐𝑒4→min$$

После подстановки регрессионных зависимостей получаем:

$$𝐻_1=1492.1775𝑁_1+5.8112⋅𝑠𝑝𝑒𝑒𝑑_1⋅𝑁_1+78.2357⋅𝑟𝑎𝑚_1⋅𝑁_1−0.5357⋅ℎ𝑑_1⋅𝑁_1$$

$$+1492.1775𝑁_2+5.8112⋅𝑠𝑝𝑒𝑒𝑑_2⋅𝑁_2+78.2357⋅𝑟𝑎𝑚_2⋅𝑁_2−0.5357⋅ℎ𝑑_2⋅𝑁_2$$

$$+1492.1775𝑁_3+5.8112⋅𝑠𝑝𝑒𝑒𝑑_3⋅𝑁_3+78.2357⋅𝑟𝑎𝑚_3⋅𝑁_3−0.5357⋅ℎ𝑑_3⋅𝑁_3$$

$$+1492.1775𝑁_4+5.8112⋅𝑠𝑝𝑒𝑒𝑑_4⋅𝑁_4
+78.2357⋅𝑟𝑎𝑚_4⋅𝑁_4−0.5357⋅ℎ𝑑_4⋅𝑁_4$$

Обозначим:
$$1492.1775N_1 = p_1,$$
$$6\cdot1492.1775\leq p_1\leq 10\cdot 1492.1775$$
$$speed_1\cdot N_1 = s_1$$
$$ram_1\cdot N_1 = r_1$$
$$ hd_1\cdot N_1 = h_1$$
___
$$1492.1775N_2 = p_2,$$
$$6\cdot 1492.1775\leq p_2\leq 22\cdot 1492.1775$$
$$speed_2\cdot N_2 = s_2$$
$$ram_2\cdot N_2 = r_2$$
$$hd_2\cdot N_2 = h_2$$
___
$$1492.1775N_3 = p_3,$$
$$11\cdot 1492.1775\leq p_3\leq 15\cdot 1492.1775$$
$$speed_3\cdot N_3 = s_3$$
$$ram_3\cdot N_3 = r_3$$
$$hd_3\cdot N_3 = h_3$$
___
$$1492.1775N_4 = p_4,$$
$$4\cdot 1492.1775\leq p_4\leq 13\cdot 1492.1775$$
$$speed_2\cdot N_4 = s_4$$
$$ram_2\cdot N_4 = r_4$$
$$hd_2\cdot N_4 = h_4$$

Для нахождения ограничений на остальные  $𝑥_i$  можно взять усредненные характеристики:

In [10]:
mean(data$speed)
mean(data$ram)
mean(data$hd)

[1] 52.01102

[1] 8.286947

[1] 416.6017

Соответственно остальные ограничения следующие:
<center>price</center>
$$6\cdot 1492.1775\leq p_{1}\leq 10 \cdot 1492.1775$$
<center>speed</center>
$$6\cdot 52.011\leq s_1\leq 10\cdot 52.011$$
<center>ram</center>
$$6\cdot 8.2869\leq r_1\leq 10\cdot 8.2869$$
<center>hd</center>
$$6\cdot 416.6017\leq h_1\leq 10\cdot 416.6017$$


---



<center>price</center>
$$6\cdot 1492.1775\leq p_{2}\leq 22\cdot 1492.1775$$
<center>speed</center>
$$6\cdot 52.011\leq s_2\leq 22\cdot 52.011$$
<center>ram</center>
$$6\cdot 8.2869\leq r_2\leq 22\cdot 8.2869$$
<center>hd</center>
$$6\cdot 416.6017\leq h_2\leq 22\cdot 416.6017$$


---



<center>price</center>
$$11\cdot 1492.1775\leq p_{3}\leq 15\cdot 1492.1775$$
<center>speed</center>
$$11\cdot 52.011\leq s_3\leq 15\cdot 52.011$$
<center>ram</center>
$$11\cdot 8.2869\leq r_3\leq 15\cdot 8.2869$$
<center>hd</center>
$$11\cdot 416.6017\leq h_3\leq 15\cdot 416.6017$$


---



<center>price</center>
$$4\cdot 1492.1775\leq p_{4}\leq 13\cdot 1492.1775$$
<center>speed</center>
$$4\cdot 52.011\leq s_4\leq 13\cdot 52.011$$
<center>ram</center>
$$4\cdot 8.2869\leq r_4\leq 13\cdot 8.2869$$
<center>hd</center>
$$4\cdot 416.6017\leq h_4\leq 13\cdot 416.6017$$


Учтем ограничения нижнего уровня:

$$p_{1} + 5.8112\cdot s_{1} + 78.2357\cdot r_{1} - 0.5357\cdot h_{1} \leq 10000$$
$$s_{2} + r_{2} \geq 300$$
$${𝑟𝑎𝑚}_3 \leq 8$$
$${hd}_4 \geq 1000$$
$$s_{4} + r_{4} \geq 800$$

Проведём нормализацию $s_2$ и $r_2$ для $С_2$ и $s_4$ и $r_4$ для $С_4$, для этого необходимо найти максимальные значения speed, ram, $N_2$ и $N_4$.

In [11]:
max(data$speed)
max(data$ram)

[1] 100

[1] 32

$$\bar{s}_{2}=\frac{s_{2}}{max(N_2)\cdot max(speed)}=\frac{s_{2}}{22\cdot 100}=\frac{s_{2}}{2200}$$
$$\bar{r}_{2}=\frac{r_{2}}{max(N_2)\cdot max(ram)}=\frac{r_{2}}{22\cdot 32}=\frac{r_{2}}{704}$$

$$\bar{s}_{4}=\frac{s_{4}}{max(N_4)\cdot max(speed)}=\frac{s_{4}}{13\cdot 100}=\frac{s_{4}}{1300}$$
$$\bar{r}_{4}=\frac{r_{4}}{max(N_4)\cdot max(ram)}=\frac{r_{4}}{13\cdot 32}=\frac{r_{4}}{416}$$

Сформулируем также частные критерии оптимизации на нижнем уровне:
Cуммарная производительность (speed, ram) в сегменте  $C_2$ и $C_4$  максимизируется.
#Для сегмента $C_2$:
$$W_2 = \alpha\cdot s_2+\beta \cdot r_2\rightarrow \max$$
Подставим нормализованные значения:
$$W_2 = \alpha\cdot \frac{s_{2}}{2200}+\beta \cdot \frac{r_{2}}{704}\rightarrow \max$$
Предположим что объем оперативной памяти и скорость имеют одинаковую важность для рабочих станций аналитиков и менеджеров, поэтому: $\alpha = 0.5, \beta = 0.5$:
$$W_2 = 0.5\cdot \frac{s_{2}}{2200} + 0.5 \cdot \frac{r_{2}}{704}\rightarrow \max$$
#Для сегмента $C_4$:
$$W_4 = \alpha\cdot s_4+\beta \cdot r_4\rightarrow \max$$
Подставим нормализованные значения:
$$W_4 = \alpha\cdot \frac{s_{4}}{1300}+\beta \cdot \frac{r_{4}}{416}\rightarrow \max$$
Предположим для станций сопряжения с видео и аудио оборудованием скорость имееет больший приоритет: $\alpha = 0.6, \beta = 0.4$:
$$W_4 = 0.6\cdot \frac{s_{4}}{1300} + 0.4 \cdot \frac{r_{4}}{416}\rightarrow \max$$
#Формируем критерий нижнего уровня:
$$H_2=0.5 W_1+0.5\cdot W_2= 0.5\cdot 0.5\cdot \frac{s_{2}}{2200} + 0.5\cdot 0.5 \cdot \frac{r_{2}}{704} + W_4 = 0.5\cdot 0.6\cdot \frac{s_{4}}{1300} + 0.5\cdot 0.4 \cdot \frac{r_{4}}{416}\rightarrow \max$$

#Окончательно получаем:
$$𝐻_1={p_{1}}+5.8112⋅{s_{1}}+78.2357{r_{1}}−0.5357⋅{h_{1}}$$

$$+{p_{2}}+5.8112⋅{s_{2}}+78.2357⋅{r_{2}}−0.5357⋅{h_{2}}$$

$$+{p_{3}}+5.8112⋅{s_{3}}+78.2357⋅{r_{3}}−0.5357⋅{h_{3}}$$

$$+{p_{4}}+5.8112⋅{s_{4}}+78.2357⋅{r_{4}}−0.5357⋅{h_{4}}$$
___
$$H_2= 0.5\cdot 0.5\cdot \frac{s_{2}}{2200} + 0.5\cdot 0.5 \cdot \frac{r_{2}}{704} + 0.5\cdot 0.6\cdot \frac{s_{4}}{1300} + 0.5\cdot 0.4 \cdot \frac{r_{4}}{416}\rightarrow \max$$
___
<center>price</center>
$$6\cdot 1492.1775\leq p_{1}\leq 10 \cdot 1492.1775$$
<center>speed</center>
$$6\cdot 52.011\leq s_1\leq 10\cdot 52.011$$
<center>ram</center>
$$6\cdot 8.2869\leq r_1\leq 10\cdot 8.2869$$
<center>hd</center>
$$6\cdot 416.6017\leq h_1\leq 10\cdot 416.6017$$

---

<center>price</center>
$$6\cdot 1492.1775\leq p_{2}\leq 22\cdot 1492.1775$$
<center>speed</center>
$$6\cdot 52.011\leq s_2\leq 22\cdot 52.011$$
<center>ram</center>
$$6\cdot 8.2869\leq r_2\leq 22\cdot 8.2869$$
<center>hd</center>
$$6\cdot 416.6017\leq h_2\leq 22\cdot 416.6017$$

---

<center>price</center>
$$11\cdot 1492.1775\leq p_{3}\leq 15\cdot 1492.1775$$
<center>speed</center>
$$11\cdot 52.011\leq s_3\leq 15\cdot 52.011$$
<center>ram</center>
$$11\cdot 8.2869\leq r_3\leq 15\cdot 8.2869$$
<center>hd</center>
$$11\cdot 416.6017\leq h_3\leq 15\cdot 416.6017$$

---

<center>price</center>
$$4\cdot 1492.1775\leq p_{4}\leq 13\cdot 1492.1775$$
<center>speed</center>
$$4\cdot 52.011\leq s_4\leq 13\cdot 52.011$$
<center>ram</center>
$$4\cdot 8.2869\leq r_4\leq 13\cdot 8.2869$$
<center>hd</center>
$$4\cdot 416.6017\leq h_4\leq 13\cdot 416.6017$$


---

$$p_{1} + 5.8112\cdot s_{1} + 78.2357\cdot r_{1} - 0.5357\cdot h_{1} \leq 10000$$
$$s_{2} + r_{2} \geq 300$$
$${𝑟𝑎𝑚}_3 \leq 8$$
$${hd}_4 \geq 1000$$
$$s_{4} + r_{4} \geq 800$$

# Оптимизация методом последовательных уступок

In [12]:
install.packages("lpSolve")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [13]:
library(lpSolve)

In [33]:
# целевая функция
objective.in <- c(1,       5.8112,      78.2357,      -0.5357,
                  1,       5.8112,      78.2357,      -0.5357,
                  1,       5.8112,      78.2357,      -0.5357,
                  1,       5.8112,      78.2357,      -0.5357)

# матрица ограничений
mat<-c(
      # h_2 -> max
      0, 0, 0, 0,
      0, 0.5 * 0.5 * 1/2200, 0.5* 0.5 * 1/ 704, 0,
      0, 0, 0, 0,
      0, 0.5 * 0.6 * 1/1300, 0.5 * 0.4 * 1/416, 0)
for (i in 1:16){
  tmp <- c()
  for (j in 1:16){
    if (j == i){
      tmp <- append(tmp, 1)
    }
    else{
      tmp <- append(tmp, 0)
    }
  }
  mat <- append(mat, tmp)
  mat <- append(mat, tmp)
}

tmp <- c(
      # производительность C_2 >= 300
      0, 0, 0, 0,
      0, 1, 1, 0,
      0, 0, 0, 0,
      0, 1, 1, 0,
      # производительность C_4 >= 800
      0, 0, 0, 0,
      0, 1, 1, 0,
      0, 0, 0, 0,
      0, 1, 1, 0,
      # стоимость <= 10000
      1, 5.8112, 78.2357, -0.5357,
      0, 0, 0, 0,
      0, 0, 0, 0,
      0, 0, 0, 0
)
mat <- append(mat, tmp)
const.mat<-matrix(mat, nrow=36, byrow=TRUE)

# ограничения
tmp <- c("<=", ">=")
signs <- c()

for (i in 1: 16){
  signs <- append(signs, tmp)
}
signs <- append(c(">="), signs)
signs <- append(signs, c(">=", ">=", "<="))
const.dir<-signs

const.rhs<-c(
    0,
    10 * 1492.1775, 6 * 1492.1775, 10 * 52.0110, 6 * 52.0110, 10 * 8.2869, 6 * 8.2869, 10 * 416.6017, 6 * 416.6017,
    22 * 1492.1775, 6 * 1492.1775, 22 * 52.0110, 6 * 52.0110, 22 * 8.2869, 6 * 8.2869, 22 * 416.6017, 6 * 416.6017,
    15 * 1492.1775, 11 * 1492.1775, 15 * 52.0110, 11 * 52.0110, 15 * 8.2869, 11 * 8.2869, 15 * 416.6017, 11 * 416.6017,
    13 * 1492.1775, 4 * 1492.1775, 13 * 52.0110, 4 * 52.0110, 13 * 8.2869, 4 * 8.2869, 13 * 416.6017, 4 * 416.6017,
    300,
    800,
    10000
)

# h минимизируется
Res<-lp("min", objective.in, const.mat, const.dir, const.rhs)
Res

Error: no feasible solution found

In [22]:
solution <- Res$solution
solution

[1]  5968.7100   208.0440    33.1476  2916.2119 19398.3075   676.1430
 [7]   107.7297  9165.2374 22382.6625   780.1650   124.3035 13747.8561
[13]  1492.1775    52.0110     8.2869  3332.8136

In [23]:
HH1 <- function(sol){
  return (sum(
    c(1,       5.8112,      78.2357,      -0.5357,
      1,       5.8112,      78.2357,      -0.5357,
      1,       5.8112,      78.2357,      -0.5357,
      1,       5.8112,      78.2357,      -0.5357) * sol
  ))
}

HH2 <- function(sol){
  return (sum(
    c(0, 0, 0, 0,
      0, 0.5 * 0.5 * 1/2200, 0.5* 0.5 * 1/ 704, 0,
      0, 0, 0, 0,
      0, 0.5 * 0.6 * 1/1300, 0.5 * 0.4 * 1/416, 0) * sol
  ))
}

H1 = HH1(Res$solution)
H2 = HH2(Res$solution)

HH1(Res$solution)
"=========="
HH2(Res$solution)

[1] 64988.78

[1] "=========="

[1] 0.1310773

In [24]:
xs = solution[c(1, 5, 9, 13)]
Ns = xs / 1492.1775
Ns

[1]  4 13 15  1